File: mobile/lib/models/game.dart
class Game {
  final String gameId;
  final List<String> board;
  final String currentPlayer;
  final String winner;
  final bool isFinished;
  String? playerX;
  String? playerO;

  Game({
    required this.gameId,
    required this.board,
    required this.currentPlayer,
    required this.winner,
    required this.isFinished,
    this.playerX,
    this.playerO,
  });

  factory Game.fromJson(Map<String, dynamic> json) {
    return Game(
      gameId: json['gameId'],
      board: List<String>.from(json['board']),
      currentPlayer: json['currentPlayer'],
      winner: json['winner'],
      isFinished: json['isFinished'],
      playerX: json['playerX'],
      playerO: json['playerO'],
    );
  }
}

File: backend/TicTacToe/TicTacToe/Models/Game.cs
namespace TicTacToe.Models;

public class Game
{
    public Guid GameId { get; set; }
    public string[] Board { get; set; } = Enumerable.Repeat("", 9).ToArray(); // 3x3
    public string CurrentPlayer { get; set; } = "X";
    public string Winner { get; set; } = "";
    public Guid? PlayerX { get; set; }
    public Guid? PlayerO { get; set; }
    public bool IsFinished => !string.IsNullOrEmpty(Winner) || Board.All(c => !string.IsNullOrEmpty(c));
}